packages/server/src/user-tokens/service.ts

```diff
@@ -94,12 +94,22 @@ export class UserTokenService extends Service {
     }
   }
 
-  async verifyToken(id: string, token: string): Promise<UserToken | undefined> {
+  async verifyToken(userId: string, rawToken: string): Promise<UserToken | undefined> {
+    const [id, token] = rawToken.split('.')
+
+    if (!validateUuid(id) || !token?.length) {
+      return undefined
+    }
+
     const userToken = await this.fetchById(id)
     if (!userToken) {
       return undefined
     }
 
+    if (userToken.userId !== userId) {
+      return undefined
+    }
+
```

Comment:
Fixed a huge security flaw. We didn't check if a user token belonged to that user! This needs to be added to the security tests. Otherwise another user can just use its own credentials to pretend to be any user of that belongs to the same clientId.